### DISCUSSION TITLE
How deep can `useDndMonitor` see?

### DISCUSSION THREAD

**Discussion by growthwp** [clauderic/dnd-kit#279] [Q&A] [answered]
I have a pretty nested structure, but at its top, there is a `DndContext`, currently, it looks like this:
```js
<DndContext>
  <Sidebar/>
  <Board>
     <Section> //is a droppable and a draggable
        <Container /> //is a droppable and a draggable
        <Container />
     </Section>
  </Board>
</DndContext>
```
The monitor is able to see events firing inside the `Section` component (which is a droppable) but isn't able to see them inside `Container`. I'm dragging an item from `Sidebar` in this case.

Could it be that the context only works with the first droppable it sees and ignores the rest?
---
Edit: My hope was that, with the introduction of `useDndMonitor`, we'd be able to see ALL drags & drops happening within the same context, but it turns out that the moment the context sees a droppable, that's where it stops.

**Comment by clauderic** [comment, marked as answer]
Do any of those components render other `<DndContext>` providers? If so, it's expected that `useDndMonitor` would only "see" within its own context provider:

```
<DndContext> // Drag start / over / end
  <Sidebar/> // `useDndMonitor` catches event
  <Board> // `useDndMonitor` catches event
     <DndContext>
        <Container /> // `useDndMonitor` does not catch event because it happened on a `<DndContext>` provider above
        <Container />
     </Section>
  </Board>
</DndContext>
```

**Reply by growthwp** [reply]
You're right, it was some arcane stuff on my end, sorry.
